helm/hpcc/templates/localroxie.yaml

```diff
@@ -121,21 +121,28 @@ spec:
 {{ include "hpcc.addUDPCertificateVolume" (dict "root" $ "name" $roxie.name "component" "localudpkey" ) | indent 6 }}
 ---
 {{- range $service := $roxie.services }}
-{{- if ne (int $service.port)  0 }}
+{{- if ne (int $service.servicePort)  0 }}
 apiVersion: v1
 kind: Service
 metadata:
   name: {{ $service.name | quote }}
   labels:
     type: roxie-service
+{{- if hasKey $service "labels" }}
+{{ toYaml $service.labels | indent 4 }}
+{{ end -}}
+{{- if hasKey $service "annotations" }}
+  annotations:
+{{ toYaml $service.annotations | indent 4 }}
+{{ end -}}
 spec:
   ports:
-  - port: {{ $service.port }}
+  - port: {{ $service.servicePort }}
     protocol: TCP
-    targetPort: {{ $service.port }}
+    targetPort: {{ $service.servicePort }}
   selector:
     roxie-server: {{ $servername | quote }}
-  type: {{ if $service.external -}} LoadBalancer {{- else -}} ClusterIP {{- end }}
+  type: {{ if $service.public -}} LoadBalancer {{- else -}} ClusterIP {{- end }}
```

Comment:
see comment in esp.yaml - looks like generate of a Service is very similar, could be commoned up.

Reply:
Done